src/containers/Node/Node.tsx

```diff
@@ -2,207 +2,207 @@ import React from 'react';
+    const nodeId = match?.params.id as string;
+    const activeTabIdFromQuery = match?.params.activeTab;
 
-    const {id: nodeId, activeTab} = match.params;
     const [{database: tenantNameFromQuery}] = useQueryParams(nodePageQueryParams);
 
+    const activeTabId = nodePageTabSchema.parse(activeTabIdFromQuery);
+
     const [autoRefreshInterval] = useAutoRefreshInterval();
-    const {currentData, isFetching, error} = nodeApi.useGetNodeInfoQuery(
-        {nodeId},
-        {pollingInterval: autoRefreshInterval},
-    );
-    const loading = isFetching && currentData === undefined;
-    const node = currentData;
+    const {
+        currentData: node,
+        isLoading,
+        error,
+    } = nodeApi.useGetNodeInfoQuery({nodeId}, {pollingInterval: autoRefreshInterval});
+
     const capabilitiesLoaded = useCapabilitiesLoaded();
     const isDiskPagesAvailable = useDiskPagesAvailable();
 
-    const {activeTabVerified, nodeTabs} = React.useMemo(() => {
-        const hasStorage = node?.Roles?.find((el) => el === STORAGE_ROLE);
+    const pageLoading = isLoading || !capabilitiesLoaded;
+
+    const isStorageNode = node?.Roles?.find((el) => el === STORAGE_ROLE);
 
-        let nodePages = hasStorage ? NODE_PAGES : NODE_PAGES.filter((el) => el.id !== STORAGE);
+    const {activeTab, nodeTabs} = React.useMemo(() => {
+        let actulaNodeTabs = isStorageNode
+            ? NODE_TABS
+            : NODE_TABS.filter((el) => el.id !== 'storage');
         if (isDiskPagesAvailable) {
-            nodePages = nodePages.filter((el) => el.id !== STRUCTURE);
+            actulaNodeTabs = actulaNodeTabs.filter((el) => el.id !== 'structure');
         }
 
-        const actualNodeTabs = nodePages.map((page) => {
-            return {
-                ...page,
-                title: page.name,
-            };
-        });
+        const actualActiveTab =
+            actulaNodeTabs.find(({id}) => id === activeTabId) ?? actulaNodeTabs[0];
+
+        return {activeTab: actualActiveTab, nodeTabs: actulaNodeTabs};
+    }, [isStorageNode, isDiskPagesAvailable, activeTabId]);
+
+    const tenantName = node?.Tenants?.[0] || tenantNameFromQuery?.toString();
 
-        let actualActiveTab = actualNodeTabs.find(({id}) => id === activeTab);
-        if (!actualActiveTab) {
-            actualActiveTab = actualNodeTabs[0];
+    React.useEffect(() => {
+        // Dispatch only if loaded to get correct node role
+        if (!isLoading) {
+            dispatch(
+                setHeaderBreadcrumbs('node', {
+                    tenantName,
+                    nodeRole: isStorageNode ? 'Storage' : 'Compute',
+                    nodeId,
+                }),
+            );
         }
+    }, [dispatch, tenantName, nodeId, isLoading, isStorageNode]);
 
-        return {activeTabVerified: actualActiveTab, nodeTabs: actualNodeTabs};
-    }, [activeTab, node, isDiskPagesAvailable]);
+    const renderHelmet = () => {
```

Comment:
what if to move every render function to a separate component (like https://github.com/ydb-platform/ydb-embedded-ui/blob/861afab08d02e8ac481ec5d1f36035bd5eed92f3/src/containers/Tablet/Tablet.tsx ) ?

Reply:
Fixed